### DISCUSSION TITLE
How to use VirtualProperty  to obtain line numbers？

### DISCUSSION THREAD

**Discussion by suxuanning** [rfoltyns/log4j2-elasticsearch#98] [Q&A]
1、How to use VirtualProperty  to obtain line numbers？
2、How should VirtualProperty be used? Can you give a simple example?

![image](https://github.com/user-attachments/assets/023f2cff-d656-4774-b831-e64c750e3b4e)
![image](https://github.com/user-attachments/assets/5ef91154-7ac5-4547-a781-8f41d8eb83e0)

**Comment by rfoltyns** [comment]
You can find a few interesting examples here: https://github.com/rfoltyns/log4j2-elasticsearch-examples.

`event:*` is not possible at the moment due to one of the limitations of `VirtualProperty` feature - no access to `event` during serialisation. Fortunately, it's fairly straightforward to implement and I will happily accepts PRs in this scope
